File: PalindromeNumber/isPalindrome.py
def isPalindrome(x: int) -> bool:
    if x < 0:
        return False

    numReversed = 0
    xCopy = x

    while xCopy > 0:
        lastDigit = xCopy % 10 # 123 % 10 = 3
        numReversed = numReversed * 10 + lastDigit
        xCopy //= 10 # 123 // 10 = 12, remove last digit
    return x == numReversed

File: PalindromeNumber/isPalindrome.ts
function isPalindrome(x: number): boolean {
    if (x < 0) {
        return false;
    }
    let reversed = 0;
    let xCopy = x;
    while (xCopy > 0) {
        let lastDigit = xCopy % 10;
        reversed = reversed * 10 + lastDigit;
        xCopy = Math.floor(xCopy / 10);
    }

    return reversed === x;
};